### PR TITLE
chore(notifications): Deprecate previous push notifications

### PR DESCRIPTION
**Pull request by cshfang**
#### Description of changes

This PR marks the previous version of Amplify Push Notifications as being on deprecation path. This only adds comments to the APIs and makes no breaking changes to the functionality itself.

#### Description of how you validated changes
Validated the deprecation messages show up in IDE
`yarn test`

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by AllanZhengYP — APPROVED** [review]

File: packages/pushnotification/src/PushNotification.ts [hunk 1]
```diff
@@ -18,6 +18,12 @@ const REMOTE_NOTIFICATION_RECEIVED = 'remoteNotificationReceived';
 const REMOTE_TOKEN_RECEIVED = 'remoteTokenReceived';
 const REMOTE_NOTIFICATION_OPENED = 'remoteNotificationOpened';
 
+/**
+ * @deprecated This package (@aws-sdk/pushnotifications) is on a deprecation path. Please see the following link for
```
Comment (jimblanc) [inline packages/pushnotification/src/PushNotification.ts:22]:
```suggestion
 * @deprecated This package (@aws-amplify/pushnotification) is on a deprecation path. Please see the following link for
```

Here and elsewhere in PR

File: packages/pushnotification/src/PushNotification.ts [hunk 2]
```diff
@@ -18,6 +18,12 @@ const REMOTE_NOTIFICATION_RECEIVED = 'remoteNotificationReceived';
...
+ * instructions on how to migrate to a new version of Amplify Push Notifications.
+ *
+ * https://docs.amplify.aws/lib/push-notifications/migrate-from-previous-version/q/platform/react-native/
```
Comment (jimblanc) [inline packages/pushnotification/src/PushNotification.ts:25]: Will we hold off on merging/releasing until the this page is available?

**Review by jimblanc — COMMENTED** [review]
Couple comments

File: packages/pushnotification/src/PushNotification.ts (hunk 1 above)
Comment (cshfang) [reply packages/pushnotification/src/PushNotification.ts:22]: Phew good catch

**Review by israx — APPROVED** [review]

File: packages/aws-amplify/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "aws-amplify",
-  "version": "5.0.22",
+  "version": "10.0.0",
```
Comment (jimblanc) [inline packages/aws-amplify/package.json:3]: This doesn't seem right

File: packages/aws-amplify/package.json [hunk 4]
```diff
@@ -48,7 +48,7 @@
     "@aws-amplify/datastore": "4.1.4",
     "@aws-amplify/geo": "2.0.22",
     "@aws-amplify/interactions": "5.0.22",
-    "@aws-amplify/notifications": "1.0.22",
+    "@aws-amplify/notifications": "10.0.0",
```
Comment (jimblanc) [inline packages/aws-amplify/package.json:51]: Ditto

File: packages/notifications/package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "@aws-amplify/notifications",
-	"version": "1.0.22",
+	"version": "10.0.0",
```
Comment (jimblanc) [inline packages/notifications/package.json:3]: Ditto

File: packages/notifications/src/common/AWSPinpointProviderCommon/index.ts [hunk 6]
```diff
@@ -174,6 +174,7 @@ export default abstract class AWSPinpointProviderCommon
 		}
 		// Update credentials
 		this.config.credentials = currentCredentials;
+		console.log(`🎯 : ${endpointId}`);
```
Comment (jimblanc) [inline packages/notifications/src/common/AWSPinpointProviderCommon/index.ts:177]: Remove this or use normal logger?

**Review by jimblanc — APPROVED** [review]

**Pull request merged**